### PR TITLE
Modifies styling left panel in Hydrator++ 

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Modifies left panel + top panel layout in hydrator++ to mimic close to mock
- Fixes bottom panel behavior
- Modifies styling for popover in left panel based on mocks

Bamboo build: http://builds.cask.co/browse/CDAP-DRC3528-4

### CODE REVIEW

File: cdap-ui/app/features/hydratorplusplus/leftpanel.less [hunk 1]
```diff
@@ -223,26 +245,30 @@ body.theme-cdap.state-hydratorplusplus-create {
...
-        &.add-plugin-template:hover, &.add-plugin-template:focus, &:hover, &:focus {
-          background-color: fade(black, 45%);
-          border: 1px dashed #999;
-          color: white;
-        }
+        // &.add-plugin-template:hover, &.add-plugin-template:focus, &:hover, &:focus {
+        //   background-color: fade(black, 45%);
+        //   border: 1px dashed #999;
+        //   color: white;
+        // }
```
Comment (elfenheart) [inline cdap-ui/app/features/hydratorplusplus/leftpanel.less]: remove?

File: cdap-ui/app/features/hydratorplusplus/leftpanel.less [hunk 2]
```diff
@@ -223,26 +245,30 @@ body.theme-cdap.state-hydratorplusplus-create {
...
         &.add-plugin-template {
-          background: transparent;
-          border: dashed 1px;
+          width: 100%;
+          text-align: center;
+          background: #098BF9;
```
Comment (elfenheart) [inline cdap-ui/app/features/hydratorplusplus/leftpanel.less]: we should extract these colors to a variable on the top, so it's easy to change

Comment (ajainarayanan) [inline cdap-ui/app/features/hydratorplusplus/leftpanel.less]: Agreed. We need to move all the colors to be generic (used under a feature). Right now I have just used straight colors. Will do this post strata for 3.4 as it involves pulling out lot of variables out of a feature (atleast from hydrator++) and then reusing variables.

**Comment by elfenheart** [comment]
:+1: LGTM

File: cdap-ui/app/features/hydratorplusplus/controllers/create/leftpanel-ctrl.js [hunk 3]
```diff
@@ -60,22 +60,8 @@ class HydratorPlusPlusLeftPanelCtrl {
...
       let getPluginTemplateNode = (type, getter) => {
-        return [
-          angular
-            .extend(
-              { pluginType: this.GLOBALS.pluginTypes[this.selectedArtifact.name][type] },
-              addPlugin
-            )
-        ]
-        .concat(this.HydratorPlusPlusLeftPanelStore[getter]());
+        return [].concat(this.HydratorPlusPlusLeftPanelStore[getter]());
```
Comment (elfenheart) [inline cdap-ui/app/features/hydratorplusplus/controllers/create/leftpanel-ctrl.js]: why do a concat empty empty array?

**Comment by elfenheart** [comment]
:+1: LGTM

**Pull request merged**